### PR TITLE
feat: ASM Deep Links for ticketId, orderId and savedCartId

### PR DESCRIPTION
**Pull request by Zeyber**
Closes: https://jira.tools.sap/browse/CXSPA-3263

Note: Missing active cart id scenario.

### CODE REVIEW

File: feature-libs/asm/components/asm-main-ui/asm-main-ui.component.ts [hunk 1]
```diff
@@ -263,8 +267,25 @@ export class AsmMainUiComponent implements OnInit, OnDestroy {
     }
   }
 
-  andThen(_options: any) {
-    //navigate to other pages
+  protected handleDeepLinkParamsAfterStartSession(
+    parameters: AsmDeepLinkParameters
+  ) {
+    if (parameters.orderId) {
```
Comment (i333055) [inline feature-libs/asm/components/asm-main-ui/asm-main-ui.component.ts:273]: maybe check `type` is more reasonable
this.asmComponentService.getSearchParameter('type')

File: feature-libs/asm/components/asm-main-ui/asm-main-ui.component.ts [hunk 2]
```diff
@@ -263,8 +267,25 @@ export class AsmMainUiComponent implements OnInit, OnDestroy {
...
+        cxRoute: 'orderDetails',
+        params: { code: parameters.orderId },
+      });
+    } else if (parameters.ticketId) {
+      // Navigate to support ticket details
+      this.routingService.go({
+        cxRoute: 'supportTicketDetails',
+        params: { ticketCode: parameters.ticketId },
+      });
+    } else if (parameters.savedCartId) {
```
Comment (i333055) [inline feature-libs/asm/components/asm-main-ui/asm-main-ui.component.ts]: see https://jira.tools.sap/browse/CXEC-22589 here only `cartId` and seems need to check
cartType = this.asmComponentService.getSearchParameter('cartType');

**Review by scarai-sap — COMMENTED** [review]
Two things may need changed.
1.we provides parameters as below for saved/activeCart
  cartId=123&type=saved
 cartId=234&type=active
Sample as https://github.tools.sap/cx-commerce/subscriptionbundles/blob/88c4df5ac5945cf022f4018df4a0f9eafab345a6/customersupportbackoffice/backoffice/testsrc/de/hybris/platform/customersupportbackoffice/widgets/sessioncontext/util/AsmUtilTest.java#L168

2. In latest code, we only use customerId, we can't emulate customer with email address(You need update your e2e).  See high security bug [CXEC-28898](https://jira.tools.sap/browse/CXEC-28898) and UT example https://github.tools.sap/cx-commerce/subscriptionbundles/blob/88c4df5ac5945cf022f4018df4a0f9eafab345a6/customersupportbackoffice/backoffice/testsrc/de/hybris/platform/customersupportbackoffice/widgets/sessioncontext/util/AsmUtilTest.java#L44

**Comment by Zeyber** [comment]
> Two things may need changed. 1.we provides parameters as below for saved/activeCart cartId=123&type=saved cartId=234&type=active Sample as https://github.tools.sap/cx-commerce/subscriptionbundles/blob/88c4df5ac5945cf022f4018df4a0f9eafab345a6/customersupportbackoffice/backoffice/testsrc/de/hybris/platform/customersupportbackoffice/widgets/sessioncontext/util/AsmUtilTest.java#L168
> 
> 2. In latest code, we only use customerId, we can't emulate customer with email address(You need update your e2e).  See high security bug [CXEC-28898](https://jira.tools.sap/browse/CXEC-28898) and UT example https://github.tools.sap/cx-commerce/subscriptionbundles/blob/88c4df5ac5945cf022f4018df4a0f9eafab345a6/customersupportbackoffice/backoffice/testsrc/de/hybris/platform/customersupportbackoffice/widgets/sessioncontext/util/AsmUtilTest.java#L44

Thanks for the notes! I was not aware that such details existed in Jira/tests. Changes done.

File: feature-libs/asm/components/asm-main-ui/asm-main-ui.component.ts [hunk 3]
```diff
@@ -181,6 +182,10 @@ export class AsmMainUiComponent implements OnInit, OnDestroy {
       }
       const parameters = {
         customerId: this.asmComponentService.getSearchParameter('customerId'),
+        orderId: this.asmComponentService.getSearchParameter('orderId'),
+        ticketId: this.asmComponentService.getSearchParameter('ticketId'),
+        cartId: this.asmComponentService.getSearchParameter('cartId'),
+        cartType: this.asmComponentService.getSearchParameter('cartType'),
```
Comment (i333055) [inline feature-libs/asm/components/asm-main-ui/asm-main-ui.component.ts:188]: seems emulated not used ？

Comment (Zeyber) [reply feature-libs/asm/components/asm-main-ui/asm-main-ui.component.ts:188]: Not in the scope of this PR

**Review by i333055 — APPROVED** [review]

File: feature-libs/asm/root/model/asm.models.ts [hunk 4]
```diff
@@ -23,3 +23,11 @@ export interface CustomerSearchOptions {
 export interface AsmUi {
   collapsed?: boolean;
 }
+
+export interface AsmDeepLinkParameters {
```
Comment (scarai-sap) [inline feature-libs/asm/root/model/asm.models.ts:27]: thanks and I'm happy we have it.

**Review by scarai-sap — APPROVED** [review]
Looks great.

**Pull request merged**